lib/vm/src/vmcontext.rs

```diff
@@ -469,207 +498,89 @@ mod test_vmglobal_definition {
+        unsafe { &mut self.storage.as_u64 }
     }
 
     /// Return a reference to the value as an f32.
-    ///
-    /// # Safety
-    ///
-    /// To be defined (TODO).
-    #[allow(clippy::cast_ptr_alignment)]
-    pub unsafe fn as_f32(&self) -> &f32 {
-        &*(self.storage.as_ref().as_ptr() as *const f32)
+    pub fn as_f32(&self) -> &f32 {
+        unsafe { &self.storage.as_f32 }
     }
 
     /// Return a mutable reference to the value as an f32.
-    ///
-    /// # Safety
-    ///
-    /// To be defined (TODO).
-    #[allow(clippy::cast_ptr_alignment)]
-    pub unsafe fn as_f32_mut(&mut self) -> &mut f32 {
-        &mut *(self.storage.as_mut().as_mut_ptr() as *mut f32)
-    }
-
-    /// Return a reference to the value as f32 bits.
-    ///
-    /// # Safety
-    ///
-    /// To be defined (TODO).
-    #[allow(clippy::cast_ptr_alignment)]
-    pub unsafe fn as_f32_bits(&self) -> &u32 {
-        &*(self.storage.as_ref().as_ptr() as *const u32)
-    }
-
-    /// Return a mutable reference to the value as f32 bits.
-    ///
-    /// # Safety
-    ///
-    /// To be defined (TODO).
-    #[allow(clippy::cast_ptr_alignment)]
-    pub unsafe fn as_f32_bits_mut(&mut self) -> &mut u32 {
-        &mut *(self.storage.as_mut().as_mut_ptr() as *mut u32)
+    pub fn as_f32_mut(&mut self) -> &mut f32 {
+        unsafe { &mut self.storage.as_f32 }
     }
 
     /// Return a reference to the value as an f64.
-    ///
-    /// # Safety
-    ///
-    /// To be defined (TODO).
-    #[allow(clippy::cast_ptr_alignment)]
-    pub unsafe fn as_f64(&self) -> &f64 {
-        &*(self.storage.as_ref().as_ptr() as *const f64)
+    pub fn as_f64(&self) -> &f64 {
+        unsafe { &self.storage.as_f64 }
     }
 
     /// Return a mutable reference to the value as an f64.
-    ///
-    /// # Safety
-    ///
-    /// To be defined (TODO).
-    #[allow(clippy::cast_ptr_alignment)]
-    pub unsafe fn as_f64_mut(&mut self) -> &mut f64 {
-        &mut *(self.storage.as_mut().as_mut_ptr() as *mut f64)
-    }
-
-    /// Return a reference to the value as f64 bits.
-    ///
-    /// # Safety
-    ///
-    /// To be defined (TODO).
-    #[allow(clippy::cast_ptr_alignment)]
-    pub unsafe fn as_f64_bits(&self) -> &u64 {
-        &*(self.storage.as_ref().as_ptr() as *const u64)
-    }
-
-    /// Return a mutable reference to the value as f64 bits.
-    ///
-    /// # Safety
-    ///
-    /// To be defined (TODO).
-    #[allow(clippy::cast_ptr_alignment)]
-    pub unsafe fn as_f64_bits_mut(&mut self) -> &mut u64 {
-        &mut *(self.storage.as_mut().as_mut_ptr() as *mut u64)
+    pub fn as_f64_mut(&mut self) -> &mut f64 {
+        unsafe { &mut self.storage.as_f64 }
     }
 
     /// Return a reference to the value as an u128.
-    ///
-    /// # Safety
-    ///
-    /// To be defined (TODO).
-    #[allow(clippy::cast_ptr_alignment)]
-    pub unsafe fn as_u128(&self) -> &u128 {
-        &*(self.storage.as_ref().as_ptr() as *const u128)
+    pub fn as_u128(&self) -> &u128 {
```

Comment:
These being safe means we could alias memory with references to the same data but different types 🤔 ; I have no idea if that's okay or not 😆 , something to look into before shipping

Reply:
Ah ok, that's no good then. I'll move the `unsafe`'s back to the function declarations.

Reply:
It can still be safe as long as it takes a `&mut self` that will guarantee exclusivity!

Reply:
Assuming sound Rust

Reply:
This is now `pub fn to_u128(&self) -> u128 {`.